Form: update editorType and editorOptions (#7545)
DevExpress/devextreme-documentation: api-reference/10 UI Components/dxForm/5 Item Types/SimpleItem/editorOptions.md

    })
    export class AppModule { }

> If you want to define multiple editors and avoid inline declarations, combine all editor types into a [union](/Documentation/Guide/Common/TypeScript_Guides/Best_Practices/#Using_Unions_Instead_of_Base_Classes). Next, assign each of your **editorOptions** an object with this union type.

Reviewer: ```suggestion
If you want to define multiple editors and avoid inline declarations, combine all editor types into a [union](/Documentation/Guide/Common/TypeScript_Guides/Best_Practices/#Using_Unions_Instead_of_Base_Classes). Then, assign an object with this union type to each of your **editorOptions* .
```

    }
    </script>

> If you use TypeScript, want to define multiple editors, and avoid inline declarations, combine all editor types into a [union](/Documentation/Guide/Common/TypeScript_Guides/Best_Practices/#Using_Unions_Instead_of_Base_Classes). Next, assign each of your **editorOptions** an object with this union type.

Reviewer: ```suggestion
If you use TypeScript, want to define multiple editors, and avoid inline declarations, combine all editor types into a [union](/Documentation/Guide/Common/TypeScript_Guides/Best_Practices/#Using_Unions_Instead_of_Base_Classes). Then, assign an object with this union type to each of your **editorOptions**.
```

    export default App;

If you use TypeScript, want to define multiple editors, and avoid inline declarations, combine all editor types into a [union](/Documentation/Guide/Common/TypeScript_Guides/Best_Practices/#Using_Unions_Instead_of_Base_Classes). Next, assign each of your **editorOptions** an object with this union type.

    <!-- tab: App.tsx -->
    import 'devextreme/dist/css/dx.light.css';
    import Form, { SimpleItem } from 'devextreme-react/form';
    import type { JSX } from 'react';
    import type { DateBoxTypes } from 'devextreme-react/date-box';
    import type { TextBoxTypes } from 'devextreme-react/text-box';

    type EditorProps = DateBoxTypes.Properties | TextBoxTypes.Properties;

    const dateBoxOptions: EditorProps = { width: '100%', value: undefined };
    const textBoxOptions: EditorProps = { mask: '+1 (X00) 000-0000', maskRules: { X: /[02-9]/ } }; 

    function App(): JSX.Element {
        return (
            <Form>
                <SimpleItem editorType="dxDateBox" editorOptions={dateBoxOptions} />
                <SimpleItem editorType="dxTextBox" editorOptions={textBoxOptions} />
            </Form>
        );
    }

    export default App;

---

> [note] Some editor properties specified in **editorOptions** can be overwritten by Form settings - for instance, labels and placeholders. The `id` and `name` attributes of the [inputAttr](/api-reference/10%20UI%20Components/dxTextBox/1%20Configuration/inputAttr.md '/Documentation/ApiReference/UI_Components/dxTextBox/Configuration/#inputAttr') property are also overwritten. The Form generates these attributes automatically and uses them for addressing the DOM elements.

Reviewer: ```suggestion
[note] Some editor properties specified in **editorOptions** can be overwritten by Form settings, for instance, labels and placeholders. The `id` and `name` attributes of the [inputAttr](/api-reference/10%20UI%20Components/dxTextBox/1%20Configuration/inputAttr.md '/Documentation/ApiReference/UI_Components/dxTextBox/Configuration/#inputAttr') property are also overwritten. The Form generates these attributes automatically and uses them to address the DOM elements.
```
